okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/SerializableErrorInterceptor.java

```diff
@@ -40,7 +40,9 @@ public Response intercept(Chain chain) throws IOException {
         if (!(response.isSuccessful() || response.code() == 101)) {
             Collection<String> contentTypes = response.headers("Content-Type");
             InputStream body = response.body().byteStream();
-            throw SerializableErrorToExceptionConverter.getException(contentTypes, response.code(), body);
+            RuntimeException runtimeException = SerializableErrorToExceptionConverter.getException(
+                    contentTypes, response.code(), body);
+            throw new RemoteIoException(runtimeException);
```

Comment:
yay! this is great. You can actually remove a chunk more complexity here - if you delete AsyncCallTag you can just do something very similar to the rethrow code in the Java8 call adapter and all the pain we've had in the last day disappears.

Reply:
Yeah getting rid of AsyncCallTag seems like it would be a nice refactor... might put this in a separate pure-refactor PR later I think